source/common/upstream/upstream_impl.cc

```diff
@@ -903,6 +903,13 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(const HostVector& new_hosts,
           hosts_changed = true;
         }
 
+        // Did the health check address change?
+        if (*(*i)->healthCheckAddress() != *host->healthCheckAddress()) {
+          // If the health check address is updated, set the current host's health check address
+          // using the updated one.
+          (*i)->setHealthCheckAddress(host->healthCheckAddress());
```

Comment:
Although this is safe from a locking perspective since HC is only on the main thread, it seems like the HC address should be an immutable property of a host? How often does it really change? Did we consider making this a thing that causes a host to get removed/re-added? (Basically having the host check pivot off of normal address and HC address)

Reply:
> should be an immutable property of a host

Yeah, on the second thought this PR seems like a quick hack for me. 

> How often does it really change? Did we consider making this a thing that causes a host to get removed/re-added?

In practice probably not that often. Will update accordingly to make it as a case of causing a host to get removed/re-added.

Reply:
Yeah, makes sense to do a full add/remove, this code is already crazy complicated ☣️